modules/distributor/search_data.go

```diff
@@ -66,6 +66,9 @@ func extractSearchData(trace *tempopb.Trace, id []byte, extractTag extractTagFun
 
 				// Collect for any spans
 				data.AddTag(search.SpanNameTag, s.Name)
+				if s.Status != nil {
+					data.AddTag(search.StatusCodeTag, strconv.FormatInt(int64(s.Status.Code), 10))
```

Comment:
Nit: this can be simplified a tiny bit:

```suggestion
					data.AddTag(search.StatusCodeTag, strconv.Itoa(int(s.Status.Code)))
```

Though I'm not sure it's worth it, under the hood this will cast back the value to `int64` before calling `FormatInt`.